File: leetcode/python/0404_sum_of_left_leaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root.left == None and root.right == None: return 0

        stk, res = [root], 0
        while len(stk) != 0:
            node = stk.pop()
            if node.left and node.left.left == None and node.left.right == None:
                res += node.left.val
            if node.left: stk.append(node.left)
            if node.right: stk.append(node.right)
        
        return res

File: leetcode/java/0404_sum_of_left_leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res = 0;
    public int sumOfLeftLeaves(TreeNode root) {
        if (root != null) {
            sumOfLeftLeaves(root.left);
            if (root.left != null && (root.left.left == null && root.left.right == null)) res += root.left.val;
            sumOfLeftLeaves(root.right);
        }

        return res;
    }
}